### PR TITLE
[Merged by Bors] - feat: expand API around analytic functions

### PR DESCRIPTION
**Pull request by sgouezel**
More basic lemmas for analytic functions (especially versions for `AnalyticOn` of lemmas we already have for `AnalyticOnNhd`).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Basic.lean [hunk 1]
```diff
@@ -669,25 +719,49 @@ theorem AnalyticOn.congr (hs : IsOpen s) (hf : AnalyticOn 𝕜 f s) (hg : s.EqOn
...
 -/
 
+/-- If a function `f` has a power series `p` on a ball within a set and `g` is linear,
+then `g ∘ f` has the power series `g ∘ p` on the same ball. -/
+theorem ContinuousLinearMap.comp_hasFPowerSeriesWithinOnBall (g : F →L[𝕜] G)
+    (h : HasFPowerSeriesWithinOnBall f p s x r) :
+    HasFPowerSeriesWithinOnBall (g ∘ f) (g.compFormalMultilinearSeries p) s x r where
+  r_le := h.r_le.trans (p.radius_le_radius_continuousLinearMap_comp _)
+  r_pos := h.r_pos
+  hasSum := fun hy h'y => by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Analytic/Basic.lean:745]:
```suggestion
  hasSum hy h'y := by
```

File: Mathlib/Analysis/Analytic/ChangeOrigin.lean [hunk 2]
```diff
@@ -284,39 +290,69 @@ theorem analyticAt_changeOrigin (p : FormalMultilinearSeries 𝕜 E F) (rp : p.r
...
+it  also admits a power series on any subball of this ball (even with a different center provided
+it belongs to `s`), given by `p.changeOrigin`. -/
+theorem HasFPowerSeriesWithinOnBall.changeOrigin (hf : HasFPowerSeriesWithinOnBall f p s x r)
+    (h : (‖y‖₊ : ℝ≥0∞) < r) (hy : x + y ∈ insert x s) :
+    HasFPowerSeriesWithinOnBall f (p.changeOrigin y) s (x + y) (r - ‖y‖₊) where
+  r_le := by
+    apply le_trans _ p.changeOrigin_radius
+    exact tsub_le_tsub hf.r_le le_rfl
+  r_pos := by simp [h]
+  hasSum := fun {z} h'z hz => by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Analytic/ChangeOrigin.lean:309]: Same

File: Mathlib/Analysis/Analytic/ChangeOrigin.lean [hunk 3]
```diff
@@ -284,39 +290,69 @@ theorem analyticAt_changeOrigin (p : FormalMultilinearSeries 𝕜 E F) (rp : p.r
...
 
+
 section
 
-variable [CompleteSpace F] {f : E → F} {p : FormalMultilinearSeries 𝕜 E F} {x y : E} {r : ℝ≥0∞}
+variable [CompleteSpace F] {f : E → F} {p : FormalMultilinearSeries 𝕜 E F} {s : Set E}
+  {x y : E} {r : ℝ≥0∞}
+
+/-- If a function admits a power series expansion `p` within a set `s` on a ball `B (x, r)`, then
+it  also admits a power series on any subball of this ball (even with a different center provided
```
Comment (jcommelin) [inline Mathlib/Analysis/Analytic/ChangeOrigin.lean:300]:
```suggestion
it also admits a power series on any subball of this ball (even with a different center provided
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**